src/Resources/Shop/PromotionsResource.php

```diff
@@ -58,13 +58,14 @@ public function applyCoupon($basketId, string $couponCode)
      */
     public function cancelCoupon($basketId, string $couponCode, $userId)
     {
-        $params = [
-            "basketId" => $basketId,
-            "couponCode" => $couponCode,
-            "userId" => $userId
-        ];
+		$params = [
+			"basketId" => $basketId,
+			"couponCode" => $couponCode,
+			"userId" => $userId,
+			"method" => 'discardcoupon'
+		];
 
-        $response = $this->handler->handle('POST', $params, 'discard');
+		$response = $this->handler->handle('POST', $params);
```

Comment:
Did you check it?
Now you miss part of URL
Please connect to Gateway and check from TEST